merlin/schema/schema.py

```diff
@@ -263,11 +268,11 @@ def with_dtype(self, dtype, is_list: bool = None, is_ragged: bool = None) -> "Co
 
     @property
     def int_domain(self) -> Optional[Domain]:
-        return self._domain() if np.issubdtype(self.dtype, np.integer) else None
+        return self._domain() if self.dtype.elemtype.value == "int" else None
```

Comment:
Might be nice to implement this logic as a convenience method on `DType` like `is_integer` 🤔